src/pages/iou/request/step/IOURequestStepConfirmation.js

```diff
@@ -0,0 +1,363 @@
+import Navigation from '@libs/Navigation/Navigation';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as TransactionUtils from '@libs/TransactionUtils';
+import personalDetailsPropType from '@pages/personalDetailsPropType';
+import reportPropTypes from '@pages/reportPropTypes';
+import {policyPropTypes} from '@pages/workspace/withPolicy';
+import useThemeStyles from '@styles/useThemeStyles';
+import * as IOU from '@userActions/IOU';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /* Onyx Props */
+    /** The personal details of the current user */
+    ...withCurrentUserPersonalDetailsPropTypes,
+
+    /** Personal details of all users */
+    personalDetails: personalDetailsPropType,
+
+    /** The policy of the report */
+    ...policyPropTypes,
+
+    /** The full IOU report */
+    report: reportPropTypes,
+
+    /** The transaction object being modified in Onyx */
+    transaction: transactionPropTypes,
+};
+const defaultProps = {
+    personalDetails: {},
+    policy: {},
+    report: {},
+    transaction: {},
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+function IOURequestStepConfirmation({
+    currentUserPersonalDetails,
+    personalDetails,
+    policy,
+    report,
+    route: {
+        params: {iouType, reportID, transactionID},
+    },
+    transaction,
+}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {windowWidth} = useWindowDimensions();
+    const {isOffline} = useNetwork();
+    const [receiptFile, setReceiptFile] = useState();
+    const receiptFilename = lodashGet(transaction, 'filename');
+    const receiptPath = lodashGet(transaction, 'receipt.source');
+    const requestType = TransactionUtils.getRequestType(transaction);
+    const headerTitle = iouType === CONST.IOU.TYPE.SPLIT ? translate('iou.split') : translate(TransactionUtils.getHeaderTitleTranslationKey(transaction));
+    const participants = useMemo(
+        () =>
+            _.chain(transaction.participants)
+                .map((participant) => {
+                    const isPolicyExpenseChat = lodashGet(participant, 'isPolicyExpenseChat', false);
+                    return isPolicyExpenseChat ? OptionsListUtils.getPolicyExpenseReportOption(participant) : OptionsListUtils.getParticipantsOption(participant, personalDetails);
+                })
+                .filter((participant) => !!participant.login || !!participant.text)
```

Comment:
This reverted https://github.com/Expensify/App/pull/32166 which caused https://github.com/Expensify/App/issues/33151